tools/system_libs.py

```diff
@@ -647,10 +647,7 @@ def check_tag():
       else:
         names = zipfile.ZipFile(fullname + '.zip', 'r').namelist()
 
-      if not (names[0].startswith(subdir + '/') or names[0].startswith(subdir + '\\')):
-        # current zip file is old, force a retrieve
-        return False
-      return True
+      return bool(re.match(subdir + r'(\\|/|$)', names[0]))
```

Comment:
I don't understand what this change does? Perhaps add a comment in the source too?

Reply:
Separate this part of the patch to #6273, I am not sure why bots are failing but that part hopefully doesn't have any issue.